### ISSUE TITLE
How does this even work?

### ISSUE THREAD

**Issue by JaneJeon** [plausible/hosting#42]
Not a self-hosting issue here, but I tried out the repo and it “just works”, but it makes me wonder how does it even work?

In particular, while you’re standing up both the postgres and clickhouse database, they’re on different containers (meaning you can’t access them using “localhost”, which is the default for both postgres and clickhouse configurations), not to mention the fact that we haven’t even _created_ the database in both postgres and clickhouse…

In short, how is the plausible instance able to actually connect to and make queries to the database containers when the configuration is seemingly wrong and it’s not actually (seemingly) connected to the database?

**Comment by SebDanielsson** [comment]
All of the containers are specified in the same compose file and can thereby share the same network. That way they can connect to each other  without opening ports etc.

**Comment by JaneJeon** [comment]
But I thought they had to connect to each other through the docker-defined network (e.g. `postgres:5432` instead of `localhost:5432`)? Sorry if this is a stupid question, but definitely interested in learning!

**Comment by cjdenio** [comment]
Docker Compose services can access each other by their service names, and Plausible is configured to connect to `plausible_db:5432` by default (https://github.com/plausible/analytics/blob/69576aa253e5dff5f6eb512be8dc2e2c13e21b43/config/runtime.exs#L56)

**Comment by JaneJeon** [comment]
Ohhhh, I see. I was under the assumption they were connecting to localhost by default. Thanks!
